control/control_containerd.go

```diff
@@ -27,23 +21,12 @@ func NewContainerd(root, address string) (*Controller, error) {
 	}
 
 	// TODO: take lock to make sure there are no duplicates
-
-	gopts := []grpc.DialOption{
-		grpc.WithBlock(),
-		grpc.WithInsecure(),
-		grpc.WithTimeout(100 * time.Second),
-		grpc.WithDialer(dialer),
-		grpc.FailOnNonTempDialError(true),
-	}
-	conn, err := grpc.Dial(dialAddress(address), gopts...)
+	client, err := containerd.New(address, containerd.WithDefaultNamespace("buildkit"))
```

Comment:
This is not equivalent. The previous one nicely produces a connection error on invalid parameters.